get_directory_sizes.py

```diff
@@ -107,7 +107,12 @@ def request_worker():
                     data = resp.json()
                     out_queue.put((path_type, path, data))
                 except Exception as exc:
-                    logging.info('Got exception %r, requeueing', exc)
+                    encoded_path = requests.compat.quote(path)
```

Comment:
I'd rather we just encode the path up in the first request on line 102. It shouldn't cause any change for paths without special characters and should fix the issues you saw. Could you try this and see if it works for you?
```python
resp = session.get('%s%s' % (storage_api_url, requests.compat.quote(path)), timeout=30)
```

Reply:
Your suggestion works perfectly.
I "push forced" commit 12473c which implements your suggestion.